db2-plugin/src/test/java/co/cask/db2/Db2SinkTestRun.java

```diff
@@ -0,0 +1,138 @@
+import co.cask.cdap.api.dataset.table.Table;
+import co.cask.cdap.etl.api.batch.BatchSink;
+import co.cask.cdap.etl.mock.batch.MockSource;
+import co.cask.cdap.etl.proto.v2.ETLPlugin;
+import co.cask.cdap.test.DataSetManager;
+import co.cask.db.batch.sink.AbstractDBSink;
+import co.cask.hydrator.common.Constants;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.sql.Connection;
+import java.sql.Date;
+import java.sql.ResultSet;
+import java.sql.Statement;
+import java.sql.Time;
+import java.sql.Timestamp;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Test for ETL using databases.
+ */
+public class Db2SinkTestRun extends Db2PluginTestBase {
+
+  @Test
+  public void testDBSink() throws Exception {
+
+    String inputDatasetName = "input-dbsinktest";
+
+    ETLPlugin sourceConfig = MockSource.getPlugin(inputDatasetName);
+    ETLPlugin sinkConfig = new ETLPlugin(
+      UI_NAME,
+      BatchSink.PLUGIN_TYPE,
+      ImmutableMap.<String, String>builder()
+        .putAll(BASE_PROPS)
+        .put(AbstractDBSink.DBSinkConfig.TABLE_NAME, "MY_DEST_TABLE")
+        .put(Constants.Reference.REFERENCE_NAME, "DBTest")
+        .build(),
+      null);
+
+    deployETL(sourceConfig, sinkConfig, DATAPIPELINE_ARTIFACT, "testDBSink");
+    createInputData(inputDatasetName);
+
+
+    try (Connection conn = createConnection();
+         Statement stmt = conn.createStatement();
+         ResultSet resultSet = stmt.executeQuery("SELECT * FROM my_table")) {
+      Set<String> users = new HashSet<>();
+      Assert.assertTrue(resultSet.next());
+      users.add(resultSet.getString("VARCHAR_COL"));
+      Assert.assertEquals(new Date(CURRENT_TS).toString(), resultSet.getDate("DATE_COL").toString());
+      Assert.assertEquals(new Time(CURRENT_TS).toString(), resultSet.getTime("TIME_COL").toString());
+      Assert.assertEquals(new Timestamp(CURRENT_TS), resultSet.getTimestamp("TIMESTAMP_COL"));
+      Assert.assertTrue(resultSet.next());
+      users.add(resultSet.getString("VARCHAR_COL"));
+      Assert.assertEquals(ImmutableSet.of("user1", "user2"), users);
+
+    }
+  }
+
+  private void createInputData(String inputDatasetName) throws Exception {
+    // add some data to the input table
+    DataSetManager<Table> inputManager = getDataset(inputDatasetName);
+    Schema schema = Schema.recordOf(
+      "dbRecord",
+      Schema.Field.of("SMALLINT_COL", Schema.of(Schema.Type.INT)),
+      Schema.Field.of("INTEGER_COL", Schema.of(Schema.Type.INT)),
+      Schema.Field.of("BIGINT_COL", Schema.of(Schema.Type.LONG)),
+      Schema.Field.of("DECIMAL_COL", Schema.of(Schema.Type.DOUBLE)),
+      Schema.Field.of("NUMERIC_COL", Schema.of(Schema.Type.DOUBLE)),
+//      Schema.Field.of("DECFLOAT_COL", Schema.of(Schema.Type.DOUBLE)),
```

Comment:
remove?

Reply:
I may have messed something up with force push, this is uncommented and used now.